characters.c

```diff
@@ -0,0 +1,96 @@
+#include "characters.h"
+
+
+// .cost is a function pointer that we are assign for each character type
+struct character_s pc = 
+{
+  .ascii = '@',
+  .cost = pc_cost,
+};
+
+void find_loco_to_place(terrain_map_t *map, Location_t *l)
+{
+    (void)tiles[TileCount];
+    do {
+        l->x = rand() % (GRID_HEIGHT -3);
+        l->y = rand() % (GRID_WIDTH -3);
+
+        if(map->grid[l->x][l->y] == pathTile.ascii) {
+        	break;
+        }
+    } while(1); 
+}
+
+uint32_t pc_cost(struct tile_s tile) 
+{
+	switch(tile.tile_idx) {
+		case Path:
+    case PokeCenter:
+    case PokeMart:
+    case Clearing:
+			return 10;
+		case TreeTile:
+    case MountainTile:
+    case Water:
+			return INT32_MAX;
+		default:
+			// should not reach here
+			return INT32_MAX;
+  	}
+}
```

Comment:
not sure if github messed up this formatting, but also do not let cases fall through it looks bad and like I always say it is better to be explicit (the only time I will say falling though is ok is the last case falling into the default but even then thats a weird exception that you shouldn't do because you can)

Reply:
also fix this in your other cost functions

Reply:
git def wiffed the formating